src/Foundation/NSObject2.cs

```diff
@@ -475,6 +475,36 @@ public virtual bool ConformsToProtocol (NativeHandle protocol)
 			if (!Runtime.DynamicRegistrationSupported)
 				return false;
 
+			// the linker/trimmer will remove the following code if the dynamic registrar is removed from the app
+			var classHandle = ClassHandle;
+			bool new_map = false;
+			lock (protocol_cache) {
```

Comment:
Theoretically, the class map and the protocol map could use separate locks, but not sure how often this is called concurrently for it to matter.

Alternatively, ConcurrentDictionary - if it's already linked in - could also be a good choice.

Reply:
We don't use `ConcurrentDictionary` anywhere else, so that would mean it could never be linked away.

Reply:
> Theoretically, the class map and the protocol map could use separate locks

Yes, but that did not help the benchmarks (which is totally not concurrent [1] since the calls all comes from the main/ui thread) and I prefer doing the simplest PR to achieve my objectives 😄

[1] but, beside my benchmark, this has to be safe to call from many threads (hence the lock)

> Alternatively, ConcurrentDictionary 

Same as @rolfbjarne mentioned, in addition to my non-concurrency comment.

Now if a different test case comes up where concurrency has a measurable impact then it would be worth to consider both suggestions.